src/workerd/api/http.c++

```diff
@@ -1002,12 +1002,15 @@ jsg::Ref<Request> Request::constructor(
         }
 
         KJ_IF_SOME(m, initDict.method) {
-          auto originalMethod = kj::str(m);
           KJ_IF_SOME(code, tryParseHttpMethod(m)) {
             method = code;
-          } else {
-            KJ_IF_SOME(code, kj::tryParseHttpMethod(toUpper(kj::mv(m)))) {
-              method = code;
+          } else KJ_IF_SOME(code, kj::tryParseHttpMethod(toUpper(m))) {
+            method = code;
+            if (!FeatureFlags::get(js).getUpperCaseAllHttpMethods()) {
+              // This is actually the spec defined behavior. We're expected to only
```

Comment:
Would you mind adding a link to the spec in here?

Reply:
Done